cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1058,16 +1058,11 @@ public void testScheduleAuth() throws Exception {
     // suspend the schedule so that it does not start running again
     scheduleManager.suspend();
 
-    // wait for scheduled runs of workflow to run to end
-    workflowManager.waitForStatus(false, 2 , 3);
-
-    // since the schedule in AppWithSchedule is to  run every second its possible that it will trigger more than one
-    // run before the schedule was suspended so check for greater than 0 rather than equal to 1
-    Assert.assertTrue(0 < workflowManager.getHistory().size());
-    // assert that all run completed
-    for (RunRecord runRecord : workflowManager.getHistory()) {
-      Assert.assertEquals(ProgramRunStatus.COMPLETED, runRecord.getStatus());
-    }
+    // Sleep for 3 seconds to wait for all workflows launched by the schedule before suspending to start
+    TimeUnit.SECONDS.sleep(3);
```

Comment:
It waits for all program runs launched by the scheduler to complete so that the namespace can be deleted after the test finishes. The sleep here waits for the program runs to appear in the program history.